### ISSUE TITLE
Error reading DNA or RNA

### ISSUE THREAD

**Issue by anhi** [BALL-Project/ball#96]
**Reported by Jerome Courcambeck on 18 Jun 38777278 20:26 UTC**
When Ball read DNA or RNA structure like pdb ID: 1RTD, 1LAE, 1ECU, 1JZC, 1F6X Ball can not built and
recognized the sugar, phosphate structure (backbone)in DNA or RNA.
Ball should read all DNA or RNA pdb format. Try to import 1A1F directly from the PDB and not from : 
\BALLView-1.2\data\structures
but from: File -- Open -- Download PDB
The difference is in the prescence one * in your include file 1A1F
Check equally the nomenclature OP1 and O1P; OP2 and O2P. Ball should recognized both.
Several Bugg, but Felicitation for this Software.

**Comment by anhi** [comment]
**Commented by dstoeckel on 13 Jul 38796303 10:13 UTC**
I just investigated the issue you described in the bug report you opened and in deed there is a problem with the naming convention.

However, this is not the whole story. The bonds of DNA (as well as the ones of Proteins) are reconstructed using a rather dumb database. Unfortunately this database is unable to recognize whether the given nucleotide is an DNA or an RNA molecule and just treats everything as DNA. Furthermore the 3' ends of a strand are not recognized such that there is a hydrogen atom missing that even the "AddHydrogens" option in BALLView cannot add. Last but not least, you will not be able to read other bases than A, T, G, C and U, since only these are in the database.
To sum up: a rewrite of the database class is needed to make it more dynamic. Thus a proper solution will not show up on a short term.

If you can live with these issues you might want to try the attached patch. It simply adds renaming rules to BALL, such that (most) of the bonds are correctly recognized.

Thank you for reporting this bug!

**Comment by anhi** [comment]
**Commented by dstoeckel on 24 Nov 39527424 15:33 UTC**
The patch has been applied to the devel and another fix ([c2cfee2ca2b72eeffd023f41be8566421dc4cb26]) by Marcel actually makes it usable, as now mixed Systems can be read in. It will appear in version 1.3 or, if another beta is necessary, 1.3-beta3.

**Comment by anhi** [comment]
**Commented by Marcel on 10 Jun 39527985 10:40 UTC**
Furthermore, patch fdadd15dd6fdaad02728e3bd9f09e0ac99db9358
now also allows to directly read RNA molecules (instead of only DNA).

**Comment by anhi** [comment]
**Commented by dstoeckel on 11 Apr 39596668 15:33 UTC**
Perhaps I was a little quick with closing this bug. RNA still remains problematic, as patch fdadd15dd6fdaad02728e3bd9f09e0ac99db9358 did introduce other regressions and my work only concerned DNA. However I think we can move this bug to another milestone as it is no regression and we are close to releasing version 1.3.

**Comment by anhi** [comment]
**Commented by akdehof on 16 Jul 40936060 10:15 UTC**
Deferred to 1.4.1

**Comment by anhi** [comment]
**Commented by wolfgang on 2 Aug 41129714 06:46 UTC**
Okay, this is somewhat mitigated by PDB consistently switching to DG,DC,DA,etc. for the DNA variants and plain G,C,A,etc. for RNA. 
Even worse though, since BALL internally still treats everything as DNA variants, this might induce a number of further read errors.
A possible fix would be to check for the file revision of PDB files which is stored in a
```
REMARK   4 1A1F COMPLIES WITH FORMAT V. 3.15, 01-DEC-08
```

At least, all the remarks should be carried over when a PDB file is saved again (#355), such that old PDB files could if not automatically be at least identified manually.
